extensions/integration-tests/src/schemaCompare.test.ts

```diff
@@ -0,0 +1,184 @@
+			let now = new Date();
+			let targetDB: string = 'ads_schemaCompare_targetDB_' + now.getTime().toString();
+
+			try {
+				let dacfxService = await azdata.dataprotocol.getProvider<azdata.DacFxServicesProvider>('MSSQL', azdata.DataProviderType.DacFxServicesProvider);
+				assert(dacfxService, 'DacFx Service Provider is not available');
+				let result = await dacfxService.deployDacpac(path.join(__dirname, 'testData/Database2.dacpac'), targetDB, true, ownerUri, azdata.TaskExecutionMode.execute);
+
+				assert(result.success === true, 'Deploy database 2 (target) should succeed');
+
+				let source: azdata.SchemaCompareEndpointInfo = {
+					endpointType: azdata.SchemaCompareEndpointType.dacpac,
+					packageFilePath: dacpac1,
+					databaseName: '',
+					ownerUri: ownerUri,
+				};
+				let target: azdata.SchemaCompareEndpointInfo = {
+					endpointType: azdata.SchemaCompareEndpointType.database,
+					packageFilePath: '',
+					databaseName: targetDB,
+					ownerUri: ownerUri,
+				};
+
+				assert(schemaCompareService, 'Schema Compare Service Provider is not available');
+				let schemaCompareResult = await schemaCompareService.schemaCompare(source, target, azdata.TaskExecutionMode.execute);
+				assertSchemaCompareResult(schemaCompareResult);
+
+				let scriptFile: string = path.join(__dirname, 'schemaCompare_DPtoDB_TestScript' + now.getTime().toString() + '.sql');
+				let status = await schemaCompareService.schemaCompareGenerateScript(schemaCompareResult.operationId, dummyDBName, scriptFile, azdata.TaskExecutionMode.execute);
+				await assertScriptGenerationResult(status, scriptFile);
+			}
+			finally {
+				utils.deleteDB(targetDB, ownerUri);
+			}
+		});
+	});
+}
+
+export function assertSchemaCompareResult(schemaCompareResult: azdata.SchemaCompareResult): void {
+	assert(schemaCompareResult.areEqual === false, `Expected: the schemas are not to be equal Actual: Equal`);
+	assert(schemaCompareResult.errorMessage === null, `Expected: there should be no error. Actual Error message: "${schemaCompareResult.errorMessage}"`);
+	assert(schemaCompareResult.success === true, `Expected: success in schema compare, Actual: Failure`);
+	assert(schemaCompareResult.differences.length === 4, `Expected: 4 differences. Actual differences: "${schemaCompareResult.differences.length}"`);
+}
+
+export async function assertScriptGenerationResult(resultstatus: azdata.ResultStatus, filepath: string): Promise<void> {
+	assert(resultstatus.success === true, `Expected: success true Actual: "${resultstatus.success}" Error Message: "${resultstatus.errorMessage}`);
+
+	let retry = 10; // file takes quite long time to get created
+	let exists = false;
+	while (retry > 0 && !exists) {
+		exists = existsSync(filepath);
+		await utils.sleep(2000);
+		retry--;
+	}
+	assert(exists, `script file ${filepath} is expected to be present`);
+	let script = readFileSync(filepath, 'utf8');
+	assert(script.length > 0, `script file should ${filepath} not be empty`);
```

Comment:
Ideally, we should do some more validation. What if the script generation failed? We can address this in another changeset.